2015/proposals/agarwal-darshan.md

```diff
@@ -0,0 +1,133 @@
+		-Extended navigation bar #221 (https://github.com/swcarpentry/amy/issues/194)
+		-Autocomplete form fields #233 (https://github.com/swcarpentry/amy/issues/233)
+		-Add CSS to make display less ugly #10 (https://github.com/swcarpentry/amy/issues/10)
+		-Headings hierarchy #99 (https://github.com/swcarpentry/amy/issues/99)
+	9)Implement backups of data on server #168 (https://github.com/swcarpentry/amy/issues/168)
+	10)Keep track of sponsorships for the workshops (Event_sponsor table #74) (https://github.com/swcarpentry/amy/issues/74)
+	11)Additional event attendance information (slots and applicants)? #64 (https://github.com/swcarpentry/amy/issues/64)
+	12)Event details page should allow validation of Eventbrite link #19 (https://github.com/swcarpentry/amy/issues/19)
+
+## Technical Details
+1) For the integration of Github or any social authentication I would be using Django Social Auth following the Django Social Auth Backend System Docs  for which it will first require much configuration after installing the App as mentioned here.
+2) Use EventBrite API for integration of Eventbrite by first getting Eventbrite key for Amy and following the Eventbrite docs (http://developer.eventbrite.com/docs/)
+3) For integration with Mailman, I will use Mailman 3 (hoping to release in April 2015), since it exposes a REST API and can use mailman.client (Python) to interact with that API and by following the doc: http://mailman-cli.readthedocs.org/ (as discussed in (as discussed in https://github.com/swcarpentry/amy/issues/42) I would be
+	-Getting the domain and creating mailing lists for each workshop as required and adding members to it.
+	-Following the syntax and direct commands of mailman.client to display lists a member is signed up to and to display members subscribed to a list.
+	-Changing the moderators and owners of a list accordingly for allowing hosts to co-manage the workshops.
+4) For the introduction of Data Tables in the Amy for the display of all the tables, as discussed and implemented in https://github.com/swcarpentry/amy/pull/134 , I will use the jQuery plugin and pass data as JSON and have Data Tables render only the data you can see. I would also try to use Data Tables server-side processing as suggested by W. Trevor King.
+5) For the implementation of Reports, to retrieve the data I will be using Django database-abstraction API to query the database and for visualisation in form of graphs I would like to use Google Charts which uses HTML5/SVG technology to visualise the data. Though I am also open to any other technology for it.
+6) UI features will be added using Bootstrap3.
+7) For implementation of backups of data on server, as discussed in https://github.com/swcarpentry/amy/issues/168 and suggested by W. Trevor King, I will implement btrfs for high-frequency backups which seems reliable enough and  every 24 hours a snapshot of database can be copied to a remote location.
```

Comment:
I like this backup approach (obviously ;), but I think implementing it is going to have to be left to someone with SSH access to amy.software-carpentry.org.  In any event, [I don't think handling this backup logic should be part of amy][1].  For example, the approach I proposed in swcarpentry/amy#168 would only need local cron jobs.

[1]: https://github.com/swcarpentry/amy/issues/168#issuecomment-73799285